oauth2/handler.go

```diff
@@ -1242,7 +1247,7 @@ func (h *Handler) oAuth2Authorize(w http.ResponseWriter, r *http.Request, _ http
 	}
 
 	authorizeRequest.SetID(acceptConsentSession.ID)
-	session, err := h.updateSessionWithRequest(ctx, acceptConsentSession, flow, r, authorizeRequest)
+	session, err := h.updateSessionWithRequest(ctx, acceptConsentSession, flow, r, authorizeRequest, nil)
```

Comment:
**Non-blocking**

I can understand the point. Not sure what's like in Golang, but usually passing a flag argument implies that the function takes too much responsibility.

What we could do is decompose the big chunky method into several smaller ones, so that logic reusability is not compromised.

Reply:
+1 on Shu's comment

Reply:
Not sure if I would call this a flag argument (unless flag=optional), I added an optional parameter to populate an existing session instead of creating a new one. 

I am not sure how I would break the logic in this function, it can simply be described as: "create a session from an accept consent request". The only other approach I could think of would be something like this:

```
// keep the function as it is
session, err := h.updateSessionWithRequest(ctx, consentSession, f, r, req)
// merge the two sessions
req.Session.Merge(session)
```
But this would complicate things and afaict it would require either to hardcode some stuff in another place OR to use some reflection to merge the 2 structs. I don't think I prefer either of these approaches, unless you have a better idea on how to do it

Reply:
@wood-push-melon @BarcoMasile i think we will take a broader review once we are feature complete

codebase is quite hard to navigate as is so making it even more convoluted doesn't seem ideal to me
if we add a `TODO` i m happy to schedule something for next pulse to analyze how to simplify/refactor this

Reply:
added a todo